src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -155,24 +159,24 @@ public sealed class StructField
             string name,
             DataType dataType,
             bool isNullable = true,
-            JObject metadata = null)
+            JsonElement? metadata = null)
         {
             Name = name;
             DataType = dataType;
             IsNullable = isNullable;
-            Metadata = metadata ?? new JObject();
+            Metadata = metadata ?? new JsonElement();
```

Comment:
Keep in mind that `new JsonElement` is not the same as `new JObject()`.

```C#
JsonElement elem = new JsonElement();
Console.WriteLine(elem.ValueKind); // undefined
Console.WriteLine(elem.ToString()); // empty string

var elem = new JObject();
Console.WriteLine(elem.Type); // object
Console.WriteLine(elem.ToString()); // {}
```

I am not sure in what context this is used yet (or what it might break), but if it's important, you likely want to create a standalone `JsonElement` from "{}" and return that.